lib/snowflake_db/connection.rb

```diff
@@ -0,0 +1,56 @@
+require 'odbc_utf8'
+require 'sequel'
+module SnowflakeDB
+  class Connection
+    attr_reader :config, :unload_target, :max_file_size
+
+    PG_INCLUDE_KEYS = %w(host dbname port user password)
+    MAX_FILE_SIZE = (5*1024*1024*1024).freeze
+
+    def initialize(config)
+      @config = config
+      @statement_timeout = config['statement_timeout']
+      @unload_target = config['unload_target']
+      @max_file_size = config['max_file_size'] || MAX_FILE_SIZE
+    end
+
+    def reconnect_on_failure(&block)
+      begin
+        return yield
+      rescue Sequel::DatabaseError => e
+        raise unless connection_expired_error?(e)
+        connection.reset
+        return yield
```

Comment:
in general, this is more idiomatic as `retry` (which will also catch the return value as well, in this case)